hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/catalog/TestHoodieHiveCatalog.java

```diff
@@ -0,0 +1,147 @@
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.catalog.exceptions.DatabaseNotExistException;
+import org.apache.flink.table.catalog.exceptions.TableAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.table.factories.FactoryUtil.CONNECTOR;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test cases for {@link HoodieHiveCatalog}.
+ */
+public class TestHoodieHiveCatalog {
+  TableSchema schema =
+      TableSchema.builder()
+          .field("name", DataTypes.STRING())
+          .field("age", DataTypes.INT())
+          .build();
+  private static HoodieHiveCatalog hoodieCatalog;
+  private final ObjectPath tablePath = new ObjectPath("default", "test");
+
+  @BeforeAll
+  public static void createCatalog() {
+    hoodieCatalog = TestHoodieCatalogUtils.createHiveCatalog();
+    hoodieCatalog.open();
+  }
+
+  @AfterEach
+  public void dropTable() throws TableNotExistException {
+    hoodieCatalog.dropTable(tablePath, true);
+  }
+
+  @AfterAll
+  public static void closeCatalog() {
+    if (hoodieCatalog != null) {
+      hoodieCatalog.close();
+    }
+  }
+
+  @Test
+  public void testCreateAndGetHoodieMORTable() throws Exception {
+    Map<String, String> originOptions = new HashMap<>();
+    originOptions.put(FactoryUtil.CONNECTOR.key(), "hudi");
+    originOptions.put(FlinkOptions.TABLE_TYPE.key(), FlinkOptions.TABLE_TYPE_MERGE_ON_READ);
+    CatalogTable table =
+        new CatalogTableImpl(schema, originOptions, "hudi table");
+    hoodieCatalog.createTable(tablePath, table, false);
+    CatalogBaseTable table1 = hoodieCatalog.getTable(tablePath);
+    assertEquals(table1.getOptions().get(CONNECTOR.key()), "hudi");
+    assertEquals(table1.getOptions().get(FlinkOptions.TABLE_TYPE.key()), FlinkOptions.TABLE_TYPE_MERGE_ON_READ);
+  }
+
+  @Test
+  public void testCreateAndGetHoodieTable() throws Exception {
+    Map<String, String> originOptions =
+        Collections.singletonMap(FactoryUtil.CONNECTOR.key(), "hudi");
+    CatalogTable table =
+        new CatalogTableImpl(schema, originOptions, "hudi table");
+    hoodieCatalog.createTable(tablePath, table, false);
+    CatalogBaseTable table1 = hoodieCatalog.getTable(tablePath);
+    assertEquals(table1.getOptions().get(CONNECTOR.key()), "hudi");
```

Comment:
We may need to inspect key options like partition keys, primary keys, preCombine keys, and custom keys.